tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,128 @@
+/*
+*Design Plan
+*First fork the mason-registry repo
+*branch the repo and add the mason.toml into Bricks/<package-name>/<version>.toml
+*add source field to <version>.toml that points to the repo of the package
+*pull request from mason-registry from branch that contaons the <version>/toml
+*need to figure out how the environment of the package where mason is invoked passes its env to mason 
+*
+*
+*/
+
+use MasonUtils;
+use Spawn;
+use FileSystem;
+use TOML;
+use MasonEnv;
+use MasonNew;
+use MasonModify;
+
+proc masonPublish(args) throws {
+  /* Implement a try catch to catch the errors from functions within this function*/
+  try! {
+    if args.size > 2 {
+      for arg in args[2..] {
+        if arg == '-h' || arg == '--help'{
+          masonPublishHelp();
+          exit(0);
+        }
+      }
+    }
+    else {
+      if isGitExist() {
+        forkMasonReg();
+        branchMasonReg();
+        var package = addPackageToBricks();
+        pullRequest(package);
+      }
+      else throw new owned MasonError("Must have package set up as git repo to publish");
+        /* throw new MasonError("Must have package set up as git repo to publ;ish...... */
+      }
+    }
+  catch e: MasonError {
+    writeln(e.message());
+    exit(1);
+  }
+}
+
+
+proc forkMasonReg(){
+  var ret = runCommand("git clone https://github.com/oplambeck/mason-registry mason-registry", true);
+  return ret;
+}
+
+proc isGitExist() {
+  var urlexist = runCommand("git config --get remote.origin.url");
+  if urlexist != ''{
+    return true;
+  }
+  else return false;
+}
+
+proc geturl() {
+  var url = runCommand("git config --get remote.origin.url",true);
+  return url;
+}
+
+proc branchMasonReg(){
+  var name = getName();
```

Comment:
Can this be const too?

Reply:
There's a handful of other `var -> const` changes possible, but not going to mark them all.